src/Test/GetTargets.hs

```diff
@@ -0,0 +1,28 @@
+module Test.GetTargets(main) where
+
+import Control.Monad
+import Development.Shake
+import Test.Type
+
+main :: IO () -> IO ()
+main _sleeper = do
+  targets <- getTargets shakeOptions rules
+  unless (targets == expected) $
```

Comment:
I have `===` in Test.Type which abstracts this and gives nice errors etc. Easier to use that.